File: src/client/java/fr/hardel/asset_editor/client/javafx/components/page/enchantment/TemplateCard.java
package fr.hardel.asset_editor.client.javafx.components.page.enchantment;

import fr.hardel.asset_editor.client.javafx.components.ui.SimpleCard;
import fr.hardel.asset_editor.client.javafx.components.ui.SvgIcon;
import javafx.geometry.Insets;
import javafx.geometry.Pos;
import javafx.scene.Node;
import javafx.scene.control.Label;
import javafx.scene.layout.HBox;
import javafx.scene.layout.Priority;
import javafx.scene.layout.Region;
import javafx.scene.layout.VBox;
import javafx.scene.paint.Color;
import net.minecraft.client.resources.language.I18n;
import net.minecraft.resources.Identifier;

/**
 * TemplateCard: SimpleCard with icon (h-16 = 64px, white) + child node (Counter),
 * and title/description below.
 * Used for maxLevel, weight, anvilCost tool cards.
 */
public final class TemplateCard extends SimpleCard {

    public TemplateCard(Identifier iconPath, String titleKey, String descriptionKey, Node child) {
        // Top row: flex items-center justify-between w-full gap-4
        SvgIcon icon = new SvgIcon(iconPath, 64, Color.WHITE);

        Region spacer = new Region();
        HBox.setHgrow(spacer, Priority.ALWAYS);

        HBox topRow = new HBox(16, icon, spacer, child);
        topRow.setAlignment(Pos.CENTER);
        topRow.setMaxWidth(Double.MAX_VALUE);

        // Bottom: title + description (mb-1 between title and desc = 4px spacing)
        Label titleLabel = new Label(I18n.get(titleKey));
        titleLabel.getStyleClass().add("template-card-title");

        VBox infoBox = new VBox(4, titleLabel);
        VBox.setMargin(infoBox, new Insets(12, 0, 0, 0)); // gap between topRow and infoBox

        if (descriptionKey != null) {
            Label descLabel = new Label(I18n.get(descriptionKey));
            descLabel.getStyleClass().add("template-card-desc");
            descLabel.setWrapText(true);
            infoBox.getChildren().add(descLabel);
        }

        contentBox.getChildren().addAll(topRow, infoBox);
    }
}

File: src/client/kotlin/fr/hardel/asset_editor/client/compose/components/ui/TemplateCard.kt
package fr.hardel.asset_editor.client.compose.components.ui

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.unit.dp
import fr.hardel.asset_editor.client.compose.StudioColors
import fr.hardel.asset_editor.client.compose.StudioTypography
import net.minecraft.resources.Identifier

@Composable
fun TemplateCard(
    iconPath: Identifier,
    title: String,
    modifier: Modifier = Modifier,
    description: String? = null,
    child: (@Composable () -> Unit)? = null
) {
    SimpleCard(modifier = modifier) {
        Row(
            horizontalArrangement = Arrangement.spacedBy(16.dp),
            verticalAlignment = Alignment.CenterVertically,
            modifier = Modifier.fillMaxWidth()
        ) {
            SvgIcon(iconPath, 64.dp, Color.White)
            Spacer(Modifier.weight(1f))
            child?.invoke()
        }

        Column(
            verticalArrangement = Arrangement.spacedBy(4.dp),
            modifier = Modifier.padding(top = 12.dp)
        ) {
            Text(
                text = title,
                style = StudioTypography.semiBold(18),
                color = StudioColors.Zinc100
            )
            if (description != null) {
                Text(
                    text = description,
                    style = StudioTypography.regular(14),
                    color = StudioColors.Zinc400
                )
            }
        }
    }
}
